module/epochs/qc_client.go

```diff
@@ -0,0 +1,169 @@
+
+	address := sdk.HexToAddress(accountAddress)
+
+	// create a new instance of flow-go-sdk client
+	flowClient, err := client.New(accessAddress, grpc.WithInsecure())
+	if err != nil {
+		return nil, fmt.Errorf("could not create flow client: %w", err)
+	}
+
+	// get account for given address
+	account, err := flowClient.GetAccount(context.Background(), address)
+	if err != nil {
+		return nil, fmt.Errorf("could not get account: %w", err)
+	}
+
+	// check if account key index within range of keys
+	if len(account.Keys) <= accountKeyIndex {
+		return nil, fmt.Errorf("given account key index is bigger than the number of keys for this account")
+	}
+
+	// construct signer for signing transactions
+	accountKey := account.Keys[accountKeyIndex]
+	signer := sdkcrypto.NewInMemorySigner(privateKey, accountKey.HashAlgo)
+
+	return &QCContractClient{
+		accessAddress:     accessAddress,
+		qcContractAddress: qcContractAddress,
+		client:            flowClient,
+		account:           account,
+		accountKeyIndex:   accountKeyIndex,
+		privateKey:        privateKey,
+		signer:            signer,
+	}, nil
+}
+
+// SubmitVote submits the given vote to the cluster QC aggregator smart
+// contract. This function returns only once the transaction has been
+// processed by the network. An error is returned if the transaction has
+// failed and should be re-submitted.
+func (c *QCContractClient) SubmitVote(ctx context.Context, vote *model.Vote) error {
+
+	// get latest sealed block to execute transaction
+	latestBlock, err := c.client.GetLatestBlock(ctx, true)
+	if err != nil {
+		return fmt.Errorf("could not get latest block from node: %w", err)
+	}
+
+	// attach submit vote transaction template and build transaction
+	tx := sdk.NewTransaction().
+		SetScript(templates.GenerateSubmitVoteScript(c.getEnvironment())).
+		SetGasLimit(1000).
+		SetReferenceBlockID(latestBlock.ID).
+		SetPayer(c.account.Address)
+
+	// add signature data to the transaction and submit to node
+	err = tx.AddArgument(cadence.NewString(hex.EncodeToString(vote.SigData)))
+	if err != nil {
+		return fmt.Errorf("could not add raw vote data to transaction: %w", err)
+	}
+
+	// sign payload using account signer
+	err = tx.SignPayload(c.account.Address, c.accountKeyIndex, c.signer)
+	if err != nil {
+		return fmt.Errorf("could not sign transaction: %w", err)
+	}
+
+	// submit signed transaction to node
+	txID, err := c.submitTx(tx)
+	if err != nil {
+		return fmt.Errorf("failed to submit transaction: %w", err)
+	}
+
+	// wait for transaction to be sealed
+	result := &sdk.TransactionResult{Status: sdk.TransactionStatusUnknown}
+	for result.Status != sdk.TransactionStatusSealed {
+		result, err = c.client.GetTransactionResult(ctx, txID)
+		if err != nil {
+			return fmt.Errorf("could not get transaction result: %w", err)
+		}
+
+		// if the transaction has expired we skip waiting for seal
+		if result.Status == sdk.TransactionStatusExpired {
+			break
```

Comment:
should we return error instead?

Reply:
Yes good point, we should return an error here. The layer above will handle retrying.
https://github.com/onflow/flow-go/blob/1e0a30e7a9118aad96a706639e37ec955f562542/module/epochs/qc_voter.go#L126-L131